wrapper/src/main/java/software/amazon/jdbc/HostSpecBuilder.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import software.amazon.jdbc.hostavailability.HostAvailability;
+import software.amazon.jdbc.hostavailability.HostAvailabilityStrategy;
+import software.amazon.jdbc.hostavailability.SimpleHostAvailabilityStrategy;
+
+public class HostSpecBuilder {
+  private String host;
+  private int port = HostSpec.NO_PORT;
+  private HostAvailability availability = HostAvailability.AVAILABLE;
+  private HostRole role = HostRole.WRITER;
+  private long weight = HostSpec.DEFAULT_WEIGHT; // Greater than or equal to 0. Healthier nodes have lower weights.
+  private Timestamp lastUpdateTime;
+  private HostAvailabilityStrategy hostAvailabilityStrategy = new SimpleHostAvailabilityStrategy();
```

Comment:
Qodana is right. Default value is always overwritten.

Reply:
Removed!